modules/core/02-client/keeper/keeper.go

```diff
@@ -147,7 +147,7 @@ func (k Keeper) IterateConsensusStates(ctx sdk.Context, cb func(clientID string,
 // GetAllGenesisClients returns all the clients in state with their client ids returned as IdentifiedClientState
 func (k Keeper) GetAllGenesisClients(ctx sdk.Context) types.IdentifiedClientStates {
 	var genClients types.IdentifiedClientStates
-	k.IterateClients(ctx, func(clientID string, cs exported.ClientState) bool {
+	k.IterateClientStates(ctx, nil, func(clientID string, cs exported.ClientState) bool {
```

Comment:
I'm not a huge fan of method signatures where explicitly passing `nil` an intended/regular workflow.  Maybe this is overkill, but I think I prefer the idea of an explicit sentinel value like

```golang
var EmptyPrefx []byte
```

And use this rather than `nil`.  A value of nil could mean a lot of different things, with a value like this it is quite obvious what we are doing.

Reply:
That sounds fine to me. I'm impartial to this decision. The SDK already makes use of nil with [iterators](https://github.com/cosmos/ibc-go/blob/main/modules/light-clients/07-tendermint/store.go#L237) so it doesn't feel out of the normal for me

 Where would you place the `EmptyPrefix`, 24-host?

Reply:
Ah I see, if it's already an existing pattern to pass nil to iterators then it probably is better to just stay consistent.

Reply:
Yup. I'm still happy to go with a sentinel value if we think that is cleaner. I guess we can use nil for now an update in the future?